### PR TITLE
SIGINT-2010 : automate QAExt and Prod release process for ADO

### PR DESCRIPTION
**Pull request by siyaramblackduck**

### CODE REVIEW

File: .github/workflows/deploy-Extension.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: deploy-synopsys-extension
+
+on:
+  workflow_dispatch:
+
+jobs:
```
Comment (lsynopsys) [inline .github/workflows/deploy-Extension.yml:6]: can we add check index.js diff here as well

File: .github/workflows/deploy-synopsys-extension.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: versioning
+        id: version-update
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUBTOKEN }}
+        run: |
+          vss_extension=$(cat vss-extension.json)
```
Comment (kiransynopsys) [inline .github/workflows/deploy-synopsys-extension.yml:23]: Can we combine first two lines 

```
vss_extension=$(cat vss-extension.json)
          extension_name=$(echo $vss_extension | jq -r '.name' ) 
```
TO extension_name=$(jq -r '.name' < vss-extension.json)

File: .github/workflows/deploy-synopsys-extension.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          extension_name=$(echo $vss_extension | jq -r '.name' )
+          echo "EXTENSION_NAME=$extension_name" >> $GITHUB_ENV
+          echo "EXTENSION NAME: "$extension_name
+          current_extension_version=$(echo $vss_extension | jq -r '.version')
```
Comment (kiransynopsys) [inline .github/workflows/deploy-synopsys-extension.yml:27]: Same as above. 

`current_extension_version=$(jq -r '.version' < vss-extension.json)
`

File: .github/workflows/deploy-Extension.yml (hunk 1 above)
Comment (siyaramblackduck) [reply .github/workflows/deploy-Extension.yml]: added

File: .github/workflows/deploy-synopsys-extension.yml (hunk 2 above)
Comment (siyaramblackduck) [reply .github/workflows/deploy-synopsys-extension.yml]: changes are done

File: .github/workflows/deploy-synopsys-extension.yml (hunk 3 above)
Comment (siyaramblackduck) [reply .github/workflows/deploy-synopsys-extension.yml]: changes are done

**Review by zaman-akib — APPROVED** [review]

**Review by kiransynopsys — APPROVED** [review]

**Review by jahid1209 — APPROVED** [review]

**Pull request merged**